examples/kubernetes/static_provisioning/secret_authentication.yaml

```
@@ -0,0 +1,89 @@
+# Secret Authentication Example
+# This example demonstrates using a Kubernetes Secret to provide S3 credentials for the Mountpoint S3 CSI Driver.
+# This authentication method is particularly useful for:
+# 1. The user wants to set their own credentials which are different than the driver level credentials
+# 2. Using different credentials for different persistent volumes
+
+# First, create a Secret containing the S3 credentials
+apiVersion: v1
+kind: Secret
+metadata:
+  name: s3-credentials
+  namespace: default
+type: Opaque
+data:
+  # Using base64 encoded values. Example:
+  # echo -n "ACCESS_KEY_ID" | base64
+  key_id: QUtJQVhYWFhYWFhYWFhYWFhY 
+  # echo -n "SECRET_ACCESS_KEY" | base64
+  access_key: U0VDUkVUWFhYWFhYWFhYWFhYWFhYWFhYWFhYWFhYWFhYWFhYWA==
+  
+  # You can also create the secret using kubectl:
+  # kubectl create secret generic s3-credentials \
+  #     --from-literal=key_id="ACCESS_KEY_ID" \
+  #     --from-literal=access_key="SECRET_ACCESS_KEY"
+  
+  # SECURITY CONSIDERATIONS:
+  # - Kubernetes Secrets are stored unencrypted in etcd by default
+  # - Consider enabling encryption at rest: https://kubernetes.io/docs/tasks/administer-cluster/encrypt-data/
+  # - Limit access to the Secret using appropriate RBAC rules
+  # - Consider using Secret rotation for production deployments
+---
+# Next, create a PersistentVolume that references the Secret
+apiVersion: v1
+kind: PersistentVolume
+metadata:
+  name: s3-pv
+spec:
+  capacity:
+    storage: 1000Gi # ignored, required
+  accessModes:
+    - ReadWriteMany
+  persistentVolumeReclaimPolicy: Retain
+  storageClassName: "" # Required for static provisioning
+  mountOptions:
+    - allow-delete
+    - force-path-style
+  csi:
+    driver: s3.csi.aws.com
```

Did you rename every aws to scality ?

```suggestion
    driver: s3.csi.scality.com
```